### PR TITLE
chore: change docs theme to sphinx-book-theme

### PR DESCRIPTION
**Pull request by johnkim-det**
## Description

Changing from `furo` theme to `sphinx-book-theme`

With Algolia search, Light theme
algolia light
With Algolia search, Dark theme
algolia dark
With fallback search, Light theme
fallback light
With fallback search, Dark theme
fallback dark

## Test Plan

Regression testing for docs layout and functionality

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

### CODE REVIEW

File: docs/README.md [hunk 1]
```diff
@@ -55,32 +55,58 @@ Since Algolia doesn't allow filtering by arbitrary tags, we also added the
...
+Our site configures canonical links to point to `/latest` all the time. This
 is necessary for optimizing SEO, and is a common practice on other docs sites
-(python standard library docs, for instance).  As a result, the Algolia crawler
+(python standard library docs, for instance). As a result, the Algolia crawler
 must be configured with `ignoreCanonicalTo: true` before it will index anything
 other than `/latest`.
+
+### Theme and customization
+
+Currently using a customized version of `sphinx-book-theme`.
```
Comment (tara-det-ai) [inline docs/README.md:90]: We've customized the `sphinx-book-theme` to suit our needs.

Replaces: `Currently using a customized version of `sphinx-book-theme`.`

**Review by tara-det-ai — APPROVED** [review]
edited the README, approving based on a demo that i saw rather than a technical review of the files

**Review by hkang1 — COMMENTED** [review]
I see that you got a the icons it and they look awesome! I left some comments about the fallback (when algolia isn't available). What's the best way to test that?

**chore:** The search and version were on the bottom of the sidebar, they just need to be moved up (before the sidebar menu content)
Screenshot 2023-03-25 at 8 59 20 PM

**Comment by johnkim-det** [comment]
@hkang1 oddly i did not see the issue with the sidebar items ordering, but i went ahead and pushed a change to list them in the desired order in `html_sidebars`. also, i think the best way to test algolia failing is by building with `html_js_files` commented out. i don't think we need any additional changes to handle that, but let me know if you see anything.

**Comment by hkang1** [comment]
> @hkang1 oddly i did not see the issue with the sidebar items ordering, but i went ahead and pushed a change to list them in the desired order in `html_sidebars`. also, i think the best way to test algolia failing is by building with `html_js_files` commented out. i don't think we need any additional changes to handle that, but let me know if you see anything.

The reordering fixed it for me. K I'll give the commenting of `html_js_files` a shot to test non-algolia path.

**Review by hkang1 — APPROVED** [review]
Hey, the code looks great! The reordering of the sidebar items fixed it for me (when I run `make live`). The commenting out of the js files works in testing the fallback case.

Think we should maybe have someone from design take a run through it for visual confirmation?

**Pull request merged**